llvm/lib/SYCLLowerIR/SYCLConditionalCallOnDevice.cpp

```diff
@@ -103,6 +103,7 @@ SYCLConditionalCallOnDevicePass::run(Module &M, ModuleAnalysisManager &) {
                          &M);
 
     NewFCaller->setCallingConv(FCaller->getCallingConv());
+    NewFCaller->setAttributes(FCaller->getAttributes());
```

Comment:
I would recommend to outline this change and associated tests into a separate PR, because that is something we should be able to merge quickly